File: Tests/cpp_turtle/src/cpp_turtle/src/subscriber.cpp
#include <memory>

#include "rclcpp/rclcpp.hpp"
#include "turtlesim/msg/pose.hpp"

using std::placeholders::_1;

class PoseSubscriber : public rclcpp::Node
{
    public:
        PoseSubscriber()
        :Node("Turtle_pose_subscriber")
        {
            subscription_ = this->create_subscription<turtlesim::msg::Pose>(
                "/turtle1/pose", 10, std::bind(&PoseSubscriber::topic_callback, this, _1)
            );
        }
    private:
        void topic_callback(const turtlesim::msg::Pose & msg)const
        {
            RCLCPP_INFO(this->get_logger(), "I heard", msg.x, msg.angular_velocity);
        }
        rclcpp::Subscription<turtlesim::msg::Pose>::SharedPtr subscription_;
};

int main(int args, char* argv[]){
    rclcpp::init(args, argv);
    rclcpp::spin(std::make_shared<PoseSubscriber>());
    rclcpp::shutdown();
    return 0;
}

File: learning/pub_sub_py/src/pub_sub/pub_sub/subscriber.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node
from turtlesim.msg import Pose

class Subscriber(Node):
    def __init__(self):
        super().__init__("custom_turtle_subscriber")
        self.subscription_ = self.create_subscription(
            Pose,
            "/turtle1/pose",
            self.listener_callback,
            10
        )

    def listener_callback(self, pose):
        self.get_logger().info(f"I listen to {pose.x} and {pose.theta}")

def main(args = None):
    rclpy.init(args = args)
    subscriber = Subscriber()
    rclpy.spin(subscriber)
    rclpy.shutdown()